### DISCUSSION TITLE
quartopub.com: Render on demand?

### DISCUSSION THREAD

**Discussion by esimms999** [quarto-dev/quarto-cli#10774] [Q&A] [answered]
### Description

Is it possible to dynamically render on quartopub.com? If so, can someone point me to the documentation about this?

My quarto site is making use of today's date in order to display some date-related ranges: 

```
```{r}
#| echo: false
today_date <- Sys.Date()
etc.
```

I am rendering this quarto website locally and then publishing to quartopub.com; this works fine. However, what I need to accomplish is to have this dynamically evaluated whenever a user accesses the site, i.e. I don't want to pre-render locally and then publish every day in order to have today's date. The documentation I have found shows two options for publishing: pre-render/publish or make use of GitHub Actions to render; the latter would still require a daily merge in order to trigger the rendering.

Thanks for any info on this!

**Comment by cscheid** [comment, marked as answer]
You need to use Shiny in these situations. quarto.pub will for the foreseeable future remain as a publishing target for static websites (effectively the output of `quarto render`).

Shinylive is an alternative that runs shiny on the browser via WASM, and that would work. But "dynamically render" is not something that Quarto does at all, let alone quarto.pub

**Reply by esimms999** [reply]
@cscheid Many thanks for the answer.
